src/bitmovin-dotnet/RestClient.cs

```diff
@@ -35,6 +37,123 @@ public RestClient(string apiKey, string apiUrl)
             };
         }
 
+#if !NET_40
+        public async Task<T> PostAsync<T>(string url, T jsonObject)
+        {
+            var uri = new Uri(_apiUrl, url);
+            var json = JObject.FromObject(jsonObject, _serializer).ToString();
+            var content = new StringContent(json, System.Text.Encoding.UTF8, "application/json");
+            var response = await _client.PostAsync(uri, content);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new BitmovinApiException(response);
+            }
+            var result = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<ResponseEnvelope<T>>(result).Data.Result;
+        }
+
+        public async Task<string> PostAndGetIdAsync<T>(string url, T jsonObject)
+        {
+            var uri = new Uri(_apiUrl, url);
+            var json = JObject.FromObject(jsonObject, _serializer).ToString();
+            var content = new StringContent(json, System.Text.Encoding.UTF8, "application/json");
+            var response = await _client.PostAsync(uri, content);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new BitmovinApiException(response);
+            }
+            var result = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<ResponseEnvelope<IdWrapper>>(result).Data.Result.Id;
```

Comment:
I'd remove the code duplication here and just use the `PostAsync` method and extract the `Id` from there